src/appengine/handlers/cron/backup.py

```diff
@@ -44,25 +49,33 @@ def get(self):
       return
 
     kinds = [
-        kind.kind_name
-        for kind in metadata.Kind.all()
-        if (not kind.kind_name.startswith('_') and
-            kind.kind_name not in EXCLUDED_MODELS)
+        kind for kind in ndb.Model._kind_map  # pylint: disable=protected-access
+        if (not kind.startswith('_') and kind not in EXCLUDED_MODELS)
     ]
+
+    app_id = utils.get_application_id()
     timestamp = datetime.datetime.utcnow().strftime('%Y-%m-%d-%H:%M:%S')
-    taskqueue.add(
-        url='/_ah/datastore_admin/backup.create',
-        method='GET',
-        target='ah-builtin-python-bundle',
-        retry_options=taskqueue.TaskRetryOptions(
-            task_retry_limit=3,
-            min_backoff_seconds=5 * 60),  # 5 minutes backoff.
-        params={
-            'filesystem': 'gs',
-            'gs_bucket_name': '%s/%s' % (gs_bucket_name, timestamp),
-            'kind': kinds
-        })
+    output_url_prefix = 'gs://%s/%s' % (gs_bucket_name, timestamp)
+    body = {
+        'output_url_prefix': output_url_prefix,
+        'entity_filter': {
+            'kinds': kinds
+        }
+    }
+
+    try:
+      request = _datastore_client().projects().export(
+          projectId=app_id, body=body)
+      response = request.execute()
```

Comment:
nit: consider just folding this into line 67 as:

```python
_datastore_client()...export(
   ....).execute()
```

since request is never used again.

Reply:
Done!